app/agents/voice/automatic/tools/breeze/configuration.py

```
@@ -284,81 +262,301 @@ async def _handle_create_surcharge_rule(params: FunctionCallParams):
+                rules_for_creation.append(create_rule)
+
+            if rules_for_creation:
+                # Add payment method details to each rule
+                for rule in rules_for_creation:
+                    if "paymentMethodType" not in rule:
+                        rule["paymentMethodType"] = payment_method_type
+                    if "paymentMethod" not in rule:
+                        rule["paymentMethod"] = payment_method
+
+                # Validate and process
+                success, result = validate_and_process_surcharge_rules(
```

Ruff (0.14.3)

Local variable `validate_and_process_surcharge_rules` referenced before assignment

(F823)